server/src/internalClusterTest/java/org/opensearch/snapshots/SegmentReplicationSnapshotIT.java

```diff
@@ -0,0 +1,279 @@
+import org.opensearch.indices.replication.common.ReplicationType;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.test.BackgroundIndexer;
+import org.opensearch.test.InternalTestCluster;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertHitCount;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class SegmentReplicationSnapshotIT extends AbstractSnapshotIntegTestCase {
+    private static final String INDEX_NAME = "test-segrep-idx";
+    private static final String RESTORED_INDEX_NAME = INDEX_NAME + "-restored";
+    private static final int SHARD_COUNT = 1;
+    private static final int REPLICA_COUNT = 1;
+    private static final int DOC_COUNT = 1010;
+
+    private static final String REPOSITORY_NAME = "test-segrep-repo";
+    private static final String SNAPSHOT_NAME = "test-segrep-snapshot";
+
+    @BeforeClass
+    public static void assumeFeatureFlag() {
+        assumeTrue("Segment replication Feature flag is enabled", Boolean.parseBoolean(System.getProperty(FeatureFlags.REPLICATION_TYPE)));
+    }
+
+    public Settings segRepEnableIndexSettings() {
+        return getShardSettings().put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT).build();
+    }
+
+    public Settings docRepEnableIndexSettings() {
+        return getShardSettings().put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT).build();
+    }
+
+    public Settings.Builder getShardSettings() {
+        return Settings.builder()
+            .put(super.indexSettings())
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, SHARD_COUNT)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, REPLICA_COUNT);
+    }
+
+    public Settings restoreIndexSegRepSettings() {
+        return Settings.builder().put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT).build();
+    }
+
+    public Settings restoreIndexDocRepSettings() {
+        return Settings.builder().put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT).build();
+    }
+
+    @Override
+    protected boolean addMockInternalEngine() {
+        return false;
+    }
+
+    public void ingestData(int docCount, String indexName) throws Exception {
+        try (
+            BackgroundIndexer indexer = new BackgroundIndexer(
+                indexName,
+                "_doc",
+                client(),
+                -1,
+                RandomizedTest.scaledRandomIntBetween(2, 5),
+                false,
+                random()
+            )
+        ) {
+            indexer.start(docCount);
+            waitForDocs(docCount, indexer);
+            refresh(indexName);
+        }
+    }
```

Comment:
nit: This is present in SegmentReplicationIT.java. Can we reuse that somehow ?

Reply:
Would rather that be a follow up, just trying to mute our flakiness right now on 2.x.

Reply:
Make sense.